django_stubs_ext/django_stubs_ext/monkeypatch.py

```diff
@@ -0,0 +1,43 @@
+from typing import Generic, List, Optional, Type, TypeVar
+
+import django
+from django.contrib.admin import ModelAdmin
+
+_T = TypeVar("_T")
+
+
+class MPGeneric(Generic[_T]):
+    """Create a data class to hold metadata about the gneric classes needing monkeypatching.
+
+    The `version` param is optional, and a value of `None` means that the monkeypatch is
+    version-independent.
+
+    This is slightly overkill for our purposes, but useful for future-proofing against any
+    possible issues we may run into with this method.
+    """
+
+    version: Optional[int]
+    cls: Type[_T]
+
+    def __init__(self, cls: Type[_T], version: Optional[int] = None) -> None:
```

Comment:
Let's follow the same format `sys.version` uses. A tuple of `int, int, int, str?`

Reply:
we can, but i think that would really complicate the logic with no added benefit. django uses semver, so only the major releases matter. this way we don't have to parse and check a version constraint

Reply:
Ok, fair enough! 👍